### PR TITLE
Add Documentation for Supported Bazel Versions

### PR DESCRIPTION
**Pull request by gnossen**
This PR:

- Adds a document capturing how to use the `grpc/grpc` repo as a Bazel dependency and which versions of Bazel are supported.
- Creates a single source of truth for supported Bazel versions (`supported_bazel_versions.txt`) and draws from this source for both the document and the distribtests.
- Adds a script to update the supported versions based on the set of released Bazel versions.

### CODE REVIEW

**Comment by gnossen** [comment]
> Looks like you're also including changes from https://github.com/grpc/grpc/pull/29175 but that change seems quite orthogonal. Can we keep both PRs separate?

Yes. The intent was just to merge this PR after #29175.

File: test/distrib/bazel/run_bazel_distrib_test.sh [hunk 1]
```diff
@@ -17,15 +17,11 @@ set -ex
...
 cd "$(dirname "$0")"
 
-# TODO(jtattermusch): make build work with bazel 2.2.0 and bazel 1.2.1 if that's reasonably simple.
-SUPPORTED_VERSIONS=(
-  "3.7.2"
-  "4.0.0"
-  "5.0.0"
-)
+# Ignore comment lines.
+VERSIONS=$(grep -v '#' supported_versions.txt)
```
[the remark this answers is not in the document]
Comment (gnossen) [reply test/distrib/bazel/run_bazel_distrib_test.sh]: It seems like comments support is more trouble than it's worth. Removed.

File: tools/buildgen/plugins/supported_bazel_versions.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+"""Retrieves supported bazel versions from plaintext file."""
+
+_SUPPORTED_VERSIONS_FILE = "test/distrib/bazel/supported_versions.txt"
+
+
+def _get_supported_bazel_versions():
+    versions = []
+    with open(_SUPPORTED_VERSIONS_FILE, "r") as f:
+        for line in f:
+            # Ignore comments
```
[the remark this answers is not in the document]
Comment (gnossen) [reply tools/buildgen/plugins/supported_bazel_versions.py]: Comment support removed.

**Comment by gnossen** [comment]
Rebased on master instead of the C++ distribtest PR to make the review cleaner.

@jtattermusch PTALA.

**Review by jtattermusch — APPROVED** [review]
Thanks!

File: test/distrib/bazel/supported_versions.txt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# TODO(jtattermusch): make build work with bazel 2.2.0 and bazel 1.2.1 if that's reasonably simple.
```
[the remark this answers is not in the document]
Comment (gnossen) [reply test/distrib/bazel/supported_versions.txt]: @jtattermusch Holding off on merge until we've resolved this comment thread.

[the remark this answers is not in the document]
Comment (jtattermusch) [reply test/distrib/bazel/supported_versions.txt]: How about `bazel/supported_versions.txt`?

**Comment by jtattermusch** [comment]
@gnossen  can you please update the "Bazel" section here as well:

https://github.com/grpc/grpc/blob/dec86eefe4ffc108ad051a71cb5b896592de7335/BUILDING.md?plain=1#L100

It currently mentions bazel 1.0 and that's way out of date. It should point to https://github.com/grpc/grpc/blob/master/doc/bazel_support.md instead.

**Comment by jtattermusch** [comment]
@veblush  please check this policy with respect to the shared C++ foundation design to make sure they match.

**Pull request merged**